rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/serialization/ExportNodeSourceToCatalogServlet.java

```diff
@@ -0,0 +1,116 @@
+        LOGGER.info("Calling POST");
+        upload(request, response);
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) {
+        upload(request, response);
+    }
+
+    private void upload(HttpServletRequest request, HttpServletResponse response) {
+        response.setContentType("text/html");
+        File nodeSourceJsonFile = null;
+        try {
+            String sessionId = null;
+            CatalogObject catalogObject = new CatalogObject();
+            for (FileItem formItem : new ServletRequestTransformer().getFormItems(request)) {
+                if (formItem.isFormField()) {
+                    if (formItem.getFieldName().equals("sessionId")) {
+                        sessionId = formItem.getString();
+                    } else if (formItem.getFieldName().equals(BUCKET_NAME_PARAM)) {
+                        catalogObject.setBucketName(formItem.getString());
+                    } else if (formItem.getFieldName().equals(NAME_PARAM)) {
+                        catalogObject.setNodeSourceName(formItem.getString());
+                    } else if (formItem.getFieldName().equals(FILE_CONTENT_PARAM)) {
+                        nodeSourceJsonFile = File.createTempFile("node-source-configuration", ".json");
+                        new PrintWriter(nodeSourceJsonFile).write(formItem.getString());
+                        formItem.write(nodeSourceJsonFile);
+                        catalogObject.setNodeSourceJsonFile(nodeSourceJsonFile);
+                    } else if (formItem.getFieldName().equals(KIND_PARAM)) {
+                        catalogObject.setKind(formItem.getString());
+                    } else if (formItem.getFieldName().equals(COMMIT_MESSAGE_PARAM)) {
+                        catalogObject.setCommitMessage(formItem.getString());
+                    } else if (formItem.getFieldName().equals(OBJECT_CONTENT_TYPE_PARAM)) {
+                        catalogObject.setObjectContentType(formItem.getString());
+                    }
+                }
+                formItem.delete();
+            }
+            CatalogRequestBuilder catalogRequestBuilder = new CatalogRequestBuilder(catalogObject);
+            String requestUri = catalogRequestBuilder.build();
+            LOGGER.info("Node Source exported to catalog using resource URI: " + requestUri);
+            catalogRequestBuilder.postNodeSourceRequestToCatalog(sessionId, requestUri, response);
+        } catch (Exception e) {
+            logErrorAndWriteResponseToClient(e, response);
+        } finally {
+            if (nodeSourceJsonFile != null) {
+                nodeSourceJsonFile.delete();
+            }
+        }
+    }
+
+    private void logErrorAndWriteResponseToClient(Exception e, HttpServletResponse response) {
+        try {
+            String errorMessage = e.getMessage().replace("<", "&lt;").replace(">", "&gt;");
```

Comment:
Are you sure that replaceAll is not needed?

Reply:
`replace` replaces all occurence. `replaceAll` does the same but can handle regex. It is definitely not needed here.

Reply:
ah, right, thanks